pfs_middleware/pfs_middleware/middleware.py

```diff
@@ -709,8 +709,18 @@ def __call__(self, req):
                     if denial_response:
                         return denial_response
 
-                # Authorization succeeded; dispatch to a helper method
+                # Authorization succeeded
                 method = req.method
+
+                # Check whether we ought to bypass. Note that swift_owner
+                # won't be set until we call authorize
```

Comment:
Hurray for functional testing! Previous version tried to do this immediately once we get into `__call__`, which flat-out didn't work.

There's still a complication, however: trying to GET an object seems to use the default storage policy, rather than the appropriate policy for the container -- pretty sure this is because `get_container_info` calls through `make_pre_authed_env` which drops in a no-op authorize callback, so we don't know that the user is a swift_owner. Eventually, this leads to us caching the container's non-existence (!) and we just assume the default storage policy...